bundle/edu.gemini.util.security/src/main/scala/edu/gemini/util/security/auth/keychain/KeySchema2.scala

```diff
@@ -0,0 +1,157 @@
+package edu.gemini.util.security.auth.keychain
+
+import edu.gemini.spModel.core.{SPProgramID,Affiliate}
+import edu.gemini.util.security.principal._
+
+import doobie.imports._
+
+import java.sql.{SQLException, Timestamp}
+import java.io.File
+import java.security.Principal
+import java.util.logging.Logger
+
+import scalaz._, Scalaz._, effect.IO
+
+object KeySchema2 extends KeyMappers2 {
+  lazy val Log: Logger = java.util.logging.Logger.getLogger(KeySchema.getClass.getName)
+
+  // H2-specific SqlState
+  val TABLE_OR_VIEW_NOT_FOUND = SqlState("42S02")
+
+  // Current schema version
+  val SchemaVersion = 1
+
+  ////// HELPERS
+
+  def info(s: String): ConnectionIO[Unit] =
+    FC.delay(Log.info(s))
+
+  def fail[A](s: String): ConnectionIO[A] =
+    FC.delay(sys.error(s))
+
+  ////// DATABASE OPERATIONS
+
+  val createSchema: ConnectionIO[Unit] = 
+    sql"""
+      create table VERSION (
+        VALUE INTEGER NOT NULL
+      );
+      create table KEYS (
+        CLASS   VARCHAR NOT NULL,
```

Comment:
We write a tag that tells you what kind of principal we're talking about. It's not the literal class name,